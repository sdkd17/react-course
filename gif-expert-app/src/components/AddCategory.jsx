import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({target}) => {
    setInputValue( target.value );
  }

  const onSubmit = (event) => {
    event.preventDefault();

    const newInputValue = inputValue.trim().toLowerCase();
    if( newInputValue.length <= 1 ) return;

    onNewCategory(newInputValue);
    setInputValue('');
    //inputValue = ''; esta mal porque no renderiza nuevamente
  }

  return (
    <form onSubmit={ (event) => onSubmit(event) }>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={ inputValue }
        onChange={ onInputChange } //evento es un parametro implicito
      />
    </form>
  )
}

AddCategory.propTypes = {
  // categories: PropTypes.array,
  onNewCategory: PropTypes.func,
}