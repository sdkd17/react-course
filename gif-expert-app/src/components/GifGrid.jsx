import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useGetGifs } from '../hooks/useGetGifs';

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useGetGifs(category);

  return (
    <>

    
      <h3>{ category }</h3>
      {isLoading && ( <h2>Cargando...</h2> )}
      <div className='card-grid'>
        {/*
          images.map( img => (
            <li key={img.id}>{ img.title }</li>
            )
          )
          Utilizar desestructuracion */
          images.map( (img) => (
              <GifGridItem 
                key={ img.id }
                {...img} //Pasaje de parametros individuales
              />
            )
          )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
