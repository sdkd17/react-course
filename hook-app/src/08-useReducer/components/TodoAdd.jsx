import { useState } from 'react';
import PropTypes from 'prop-types';

export const TodoAdd = ({ onNewTodo }) => {

  const [input, setInput] = useState('');

  const onInputChange = ({target}) => {
    setInput(target.value);
  }

  const onSubmitEvent = (event) => {
    event.preventDefault();
    if ( input.length <= 1 ) return;  
    
    onNewTodo(input);    
    setInput('');
  }

  return (
    <form onSubmit={ onSubmitEvent }>
      <input 
        type="text"
        name="todo"
        placeholder="Â¿Que hay que hacer?"
        className="form-control"
        value={ input }
        onChange={ onInputChange }
      />

      <button
        type="submit"
        className="btn btn-outline-primary mt-2"
      >
        Agregar
      </button>
    </form>
  )
}

TodoAdd.propTypes = {
  onNewTodo: PropTypes.func.isRequired,
}
