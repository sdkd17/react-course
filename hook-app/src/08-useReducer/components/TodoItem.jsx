import PropTypes from 'prop-types';

export const TodoItem = ({ item, onDeleteItem, onDoneItem }) => {
  
  const { todo, id, done } = item;

  return (
    <li className={`list-group-item d-flex justify-content-between ${ done? 'list-group-item-success' : ''}`}>
      <span 
        className="align-self-center"
        aria-label='span-label'
      >
        { todo }
      </span>
      <div className="btn-group">
        <button
          className={"btn btn-secondary"}
          onClick={ () => onDoneItem( id ) }
        >
          {done?'undo':'done'}
        </button>
        <button
          className="btn btn-danger"
          onClick={ () => onDeleteItem( id ) }
        >
          Borrar
        </button>
      </div>
      
    </li>          
  )
}

TodoItem.propTypes = {
  item: PropTypes.object.isRequired,
  onDeleteItem: PropTypes.func.isRequired,
  onDoneItem: PropTypes.func.isRequired,
}
