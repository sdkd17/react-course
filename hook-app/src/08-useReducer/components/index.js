
export * from './TodoList';
export * from './TodoAdd';