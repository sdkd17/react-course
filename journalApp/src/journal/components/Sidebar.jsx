import { TurnedInNot } from "@mui/icons-material"
import { Grid, Drawer, Box, Toolbar, Typography, Divider, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"


export const Sidebar = ({ drawerWidth = 240}) => {
  return (
    <Box
      component='nav'
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm:0 } }}
    >
      <Drawer
        variant='permanent'
        open
        sx={{ display: { xs: 'block' },
            '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
        }}
      >
        <Toolbar>
          <Typography variant='h6' noWrap component='div'>
            Nombre Apellido
          </Typography>
        </Toolbar>

        <Divider />

        {
          ['Enero', 'Febrero', 'Marzo', 'Abril'].map( text => (
            <ListItem key={text} disablePadding>
              <ListItemButton>
                <ListItemIcon>
                  <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                  <ListItemText primary='text'/>
                  <ListItemText primary='asdasdas a asd asdasd asd asd asd asd'/>
                </Grid>
              </ListItemButton>
            </ListItem>
          ) )
        }
      </Drawer>
    </Box>
  )
}
