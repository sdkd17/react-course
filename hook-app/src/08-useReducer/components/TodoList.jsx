
import PropTypes from 'prop-types';
import { TodoItem } from './TodoItem';

export const TodoList = ({ list, onDeleteItem, onDoneItem } ) => {
  return (
    <ul className="list-group">
      {
        list.map( item => ( 
          <TodoItem 
            key={ item.id }   
            item={ item } 
            onDeleteItem={ onDeleteItem }
            onDoneItem={ onDoneItem }
          /> )
        )
      }
    </ul>  
  )
}

TodoList.propTypes = {
  list: PropTypes.array.isRequired,
  onDeleteItem: PropTypes.func.isRequired,
  onDoneItem: PropTypes.func.isRequired,
}
