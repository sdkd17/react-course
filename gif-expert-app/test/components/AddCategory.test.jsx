import { fireEvent, render, screen } from '@testing-library/react';
import { AddCategory } from '../../src/components/AddCategory';

describe('Pruebas en componente AddCategory', () => { 

  test('should match with snapshot', () => { 

    const { container } = render( <AddCategory onNewCategory={() => {}} />);
    expect( container ).toMatchSnapshot()
  });

  test('should change value on input', () => { 

    render( <AddCategory onNewCategory={() => {}} />);
    
    const input = screen.getByRole('textbox');
    //simula evento de escribir en el input, el segundo parametro 
    // tiene la misma estructura que cuando se llama a onChange definido en componente
    fireEvent.input( input, { target: { value: 'nba'} } )
    //esto chequea de que se esta cambiando el estado de inputValue, 

    expect( input.value ).toBe('nba');
  });

  test('should call onNewCategory if input has some value with length >= 1', () => { 

    const inputValue = 'nba';
    const onNewCategory = jest.fn(); //mock

    render( <AddCategory onNewCategory={ onNewCategory } />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form')

    //disparo simulacion de eventos
    fireEvent.input( input, { target: { value: inputValue } } );
    fireEvent.submit( form )

    expect( input.value ).toBe('');
    //el mock fue llamado con el parametro inputValue
    expect( onNewCategory ).toHaveBeenCalledWith( inputValue );
  });

  test('should not call onNewCategory if input has length <= 1', () => { 

    const inputValue = 'n';
    const onNewCategory = jest.fn(); //mock

    render( <AddCategory onNewCategory={ onNewCategory } />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form')

    //disparo simulacion de eventos
    fireEvent.input( input, { target: { value: inputValue } } );
    fireEvent.submit( form )

    expect( input.value ).toBe( inputValue );
    //el mock fue llamado con el parametro inputValue
    expect( onNewCategory ).toHaveBeenCalledTimes( 0 );
    // expect( onNewCategory ).not.toHaveBeenCalledTimes( 0 );
  });


})