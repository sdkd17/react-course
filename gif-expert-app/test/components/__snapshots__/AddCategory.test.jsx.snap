// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AddCategory should match with snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
